### PR TITLE
Add explicit use of 32-bit int in fast_exp

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Implements @blochl [advice](https://github.com/scikit-image/scikit-image/pull/4322#discussion_r353567366).

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/_shared/fast_exp.h [hunk 1]
```diff
@@ -3,8 +3,10 @@
  * Reference [2]: https://doi.org/10.1162/089976600300015033
  * Additional improvements by Leonid Bloch. */
 
+#include <stdint.h>
+
 /* use just EXP_A = 1512775 for integer version, to avoid FP calculations */
-#define EXP_A (1512775.3951951856938)  /* 2^20*ln2 */
+#define EXP_A (1512775.3951951856938)  /* 2^20/ln2 */
```
Comment (blochl) [inline skimage/_shared/fast_exp.h]: Since this fix is unrelated, please put it to a separate commit. It will make things easier if the functional part of this commit will ever need to be reverted, and in general making the commits as specific as possible is a good practice.

**Comment by blochl** [comment]
@rfezzani, thanks, but what you have done now is introduce two completely redundant commits to the code: the original commit, then the revert of the original commit, and then the two correct commits. This is a clutter for the Git tree. Please read about [editing commits](https://stackoverflow.com/questions/1186535/how-to-modify-a-specified-commit) if you feel like it.

What you can do now, is `git rebase -i 7a3e0d7902^` (note the `^` in the end) and in the editor that opens automatically, delete the two lines which correspond to the first commit and its revert - these should be the first two lines, save and close. Then verify that they are indeed gone in your branch (`git log --graph --oneline --decorate --all`, or `git log --graph --oneline --decorate`) and then **force** push your branch for modifying this PR.

**Comment by hmaarrfk** [comment]
how was this working before?

can we create a test for this if we enable aarch64 on travis?>?

**Comment by rfezzani** [comment]
Hi @hmaarrfk, in #4322, I added a `fast_exp.pyx` file for which I planed to add tests. Do you prefer this to be done here?

**Comment by blochl** [comment]
> how was this working before?
> 
> can we create a test for this if we enable aarch64 on travis?>?

int is 32-bit by default on most platforms, on aarch64 as well.

**Comment by hmaarrfk** [comment]
hmm, ok you are right! i guess i forgot that..... so confusing

File: skimage/_shared/fast_exp.h (hunk 1 above)
Comment (hmaarrfk) [reply skimage/_shared/fast_exp.h]: We squash many PRs. @rfezzani if you want us to merge each commit, you will have to rebase and clean the history a little as suggested by blochl

Comment (rfezzani) [reply skimage/_shared/fast_exp.h]: I don't really care if the PR is squashed. I applied blochl suggestion to not be lazy :-)

Comment (blochl) [reply skimage/_shared/fast_exp.h]: Thanks!!
Not sure if squashing the entire PR is a good idea for later references/fixes/feature history, but hey - not my call. :)

**Comment by hmaarrfk** [comment]
I guess you can do what you want in terms of incorporating tests. IMO, this is OK to merge as is.

**Comment by jni** [comment]
I'm curious about why this is needed. In general, my preference is to let the compiler/OS choose the "best" int type for the platform, unless there is good reason to explicitly choose one type over the other. Sorry, I haven't been following the full discussion from the beginning, but I tried following a few links to the previous discussion and didn't see any comment that properly explained why int32 should be the appropriate data type here...?

@blochl I admire your zeal for clean commit histories, but we are generally quite lax at skimage with it, and prefer to squash (thinking that a PR itself should be *reasonably* self-contained) rather than force contributors to jump through many git hoops. Many newcomers might have found your git instructions quite intimidating!

Thanks everyone for all the work so far on this issue, it is impressive!

**Comment by rfezzani** [comment]
Hi @jni, 32-bit integer are required for the method to work(see the reference [here](https://schraudolph.org/pubs/Schraudolph99.pdf)), it's part of the trick ;-)

**Comment by hmaarrfk** [comment]
Yeah. I read the reference a while back. The trick is actually really useful for 8 bit integers.

**Comment by jni** [comment]
Very good, thank you for the pointers! =)
